### PR TITLE
feat(multiphase): Add phase_count property to envoy

### PR DESCRIPTION
**Pull request by catsmanac**
First step to implement multi phase support. This PR prepares the way by obtaining phase configuration

- Collect CT Meter configuration from ivp/meters and determine number of phases in use. Set (new) SupportedFeatures `DUALPHASE` or `THREEPHASE` when CT's are enabled and  multiple phases are in use.
- Add new property phase_count to Envoy that will return 1,2 or 3 based on added features.
- Add test fixture 7.6.175_with_cts_3phase based on 7.6.175_with_cts and 3 phase configuration in ivp/meters and production data. To be used for testing multiphase support.
- Add ivp/meters mock to test_endpoints.py. In test_with_7_x_firmware add appropriate DUALPHASE or THREEPHASE SupportedFeatures and phase_count for each of the test fixtures.
- Add assert for phase_count in the test_with_5_0_49_firmware and V7 parameterized tests.
- Code coverage of all added/changed code with new tests is reported locally as 100%
- Snapshot updated with new 7.6.175_with_cts_3phase fixture

Code example for use by client. Phase_count >1 or added properties to be used for multi phase cases, currently information only.

```
from pyenphase.envoy import Envoy, SupportedFeatures

envoy = Envoy(host)
await envoy.setup()
await envoy.authenticate(username=username, password=password, token=token)
await envoy.update()

print(f'Phase Count {envoy.phase_count} \
            THREEPHASE: {(envoy.supported_features & SupportedFeatures.THREEPHASE)!=0} \
            DUALPHASE: {(envoy.supported_features & SupportedFeatures.DUALPHASE)!=0}'
)
```
`Phase Count 3      THREEPHASE: True       DUALPHASE: False`

Let me know what documentations needs update as well.

### CODE REVIEW

File: src/pyenphase/updaters/meters.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        # can't get to envoy from here to store a phase_count so use DUAL and THREEPHASE features
+        phase_count: int = 1
+        for meter in meters_json:
+            if meter["state"] == "enabled":
+                # we could set SupportedFeatures.METERING as well
+                phase_count = (
+                    int(meter["phaseCount"])
+                    if int(meter["phaseCount"]) > phase_count
+                    else phase_count
+                )
```
Comment (bdraco) [inline src/pyenphase/updaters/meters.py:43]: Please do the int conversion only once

**Comment by catsmanac** [comment]
This is also a first step in solving #69

File: src/pyenphase/envoy.py [hunk 2]
```diff
@@ -59,6 +60,7 @@
     EnvoyApiV1ProductionInvertersUpdater,
     EnvoyEnembleUpdater,
     EnvoyTariffUpdater,
+    EnvoyMetersUpdater,
```
[the remark this answers is not in the document]
Comment (catsmanac) [reply src/pyenphase/envoy.py:63]: Not an issue in this stage. We can do it now or in next PR (had this one already committed when I hit some complications and asked by question #104.)

**Review by bdraco — APPROVED** [review]
Tested on 3.x firmware as well to make sure it didn't cause an auth issue

**Comment by bdraco** [comment]
Thanks @catsmanac

**Pull request merged**